### PR TITLE
Support kube-scheduler config

### PR DESCRIPTION
**Pull request by justinsb**
Users can provider a kube-scheduler configuration file in the additional objects for the cluster.   The kops CLI will create one if it is not provided, and will merge in any scheduler settings in the Cluster object.  This is passed to nodeup as a static file.

Issue #13352

### CODE REVIEW

**Comment by olemarkus** [comment]
Should we validate the content of the scheduler config?

**Comment by olemarkus** [comment]
In the past we have said we may want to add support for a label to scheduler config indicating kops should be hands-off. But we need objectmeta support first.

For most users, we need to at minimum ensure they don't set things like kubeconfig path as it needs to match where nodeup actually writes it. The finished config should probably pass through the upstream validation to avoid validation error happening nodeup-time.

**Comment by justinsb** [comment]
/retest

Weird JWT/auth issues from gcloud

**Comment by justinsb** [comment]
/retest

"Your credentials are invalid" during artifact upload, seems unrelated.

**Comment by avdhoot** [comment]
@olemarkus this blocking us from upgrading our cluster because we pass scheduler config. It is sad to see this not part kops 1.23/1.24. We are breaking features in the release which is not good :(. Let me know if we can help somehow.

**Comment by olemarkus** [comment]
Well. It's not part of kops 1.23/1.24 because no one contributed the code for it. Or showed up in office hours to discuss them. Or created issues that they participated in.

Right now, the best way to contribute is to test this PR and improve on it.

**Comment by justinsb** [comment]
What do we think - are we OK to merge this now?

cc @olemarkus / @hakman

**Comment by olemarkus** [comment]
Can we have some user documentation for this? Possibly release notes as well?

Other than that, it looks good to me.

File: pkg/assets/builder.go [hunk 1]
```diff
@@ -50,15 +50,31 @@ type AssetBuilder struct {
 	// KubernetesVersion is the version of kubernetes we are installing
 	KubernetesVersion semver.Version
 
-	// StaticManifests records static manifests
+	// StaticManifests records manifests used by nodeup:
+	// * e.g. sidecare manifests for static pods run by kubelet
```
Comment (hakman) [inline pkg/assets/builder.go:54]:
```suggestion
	// * e.g. sidecar manifests for static pods run by kubelet
```

**Comment by hakman** [comment]
> Can we have some user documentation for this? Possibly release notes as well?

Same feedback from me, otherwise LGTM.

**Comment by justinsb** [comment]
Now with added docs :-)

**Review by hakman — APPROVED** [review]

**Comment by hakman** [comment]
> /hold for the 1.24 to branch of

The 1.24 branch has been created.
/hold cancel

**Comment by avdhoot** [comment]
Thank you all :pray: for your efforts. looking forward to see this change in kops 1.25

**Pull request merged**